experimental/yarpl/include/reactivestreams/ReactiveStreams.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <exception>
+#include <memory>
+
+namespace reactivestreams_yarpl {
+// TODO remove 'yarpl' if we agree on this usage elsewhere
+
+class Subscription {
+ public:
+  virtual ~Subscription() = default;
+  virtual void request(int64_t n) = 0;
+  virtual void cancel() = 0;
+};
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+  virtual void onSubscribe(Subscription*) = 0;
+  virtual void onNext(const T&) = 0;
```

Comment:
Here's how it looks implementing an operator right now:

```cpp
  void onNext(const T& t) {
    downstream_->onNext((*f_)(t));
  }

  void onNext(T&& t) {
    downstream_->onNext((*f_)(std::move(t)));
  }
```

This works and correctly calls either the lvalue or rvalue version depending on what I pass in from the top.